### PR TITLE
feat: Added assignments in credits_available endpoint

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Description**: This PR adds the learner content assignments to the [credits_available endpoint](http://localhost:18270/api/v1/policy-redemption/credits_available/?enterprise_customer_uuid=d1303765-90f9-4e01-9eaa-654c943fbd60&lms_user_id=3)

**JIRA** : ENT-7870

**Sample Output**

image

### CODE REVIEW

**Review by jajjibhai008 — APPROVED** [review]
Please add the response details in the PR description. Other than that LGTM 👍

File: enterprise_access/apps/api/serializers/subsidy_access_policy.py [hunk 1]
```diff
@@ -522,14 +522,31 @@ class SubsidyAccessPolicyCreditsAvailableResponseSerializer(SubsidyAccessPolicyR
...
         source='subsidy_expiration_datetime',
     )
+    learner_content_assignments = serializers.SerializerMethodField('get_assignments_serializer')
+
+    def get_assignments_serializer(self, obj):
+        """
+        Return serialized assignments if the policy access method is of the 'assigned' type
+        """
+        if obj.access_method == AccessMethods.ASSIGNED:
+            assignments = obj.assignment_configuration.assignments.all().filter(
```
Comment (iloveagent57) [inline enterprise_access/apps/api/serializers/subsidy_access_policy.py:532]: Two minor things:
1. Nit: we don't need the `all()` before the `filter()` here :) 
2. Performance thing: the `LearnerContentAssignmentResponseSerializer` also serializes the actions related to each serialized assignment, so we could throw a `.prefetch('actions')` onto this queryset to help keep the number of queries for actions a constant (one).

File: enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py [hunk 2]
```diff
@@ -1321,6 +1325,70 @@ def test_credits_available_endpoint(
...
+        assignment1 = LearnerContentAssignmentFactory.create(
+            assignment_configuration=assignment_configuration,
+            learner_email='alice@foo.com',
+            lms_user_id=1234,
+            content_key=content_key,
+            content_title=content_title,
+            content_quantity=-content_price_cents,
+            state=LearnerContentAssignmentStateChoices.ALLOCATED,
+        )
+        LearnerContentAssignmentFactory.create(
```
Comment (iloveagent57) [inline enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py:1362]: Maybe also worth adding a `LearnerContentAssignmentAction` record into the mix, too.

File: enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py [hunk 3]
```diff
@@ -1321,6 +1325,71 @@ def test_credits_available_endpoint(
...
+
+        query_params = {
+            'enterprise_customer_uuid': str(self.enterprise_uuid),
+            'lms_user_id': 1234,
+        }
+        response = self.client.get(self.subsidy_access_policy_credits_available_endpoint, query_params)
+
+        response_json = response.json()
+        self.assertEqual(len(response_json[0]['learner_content_assignments']), 1)
+        self.assertEqual(response_json[0]['learner_content_assignments'][0]['uuid'], str(assignment1.uuid))
```
Comment (iloveagent57) [inline enterprise_access/apps/api/v1/tests/test_subsidy_access_policy_views.py:1391]: We probably want to do some validation of what's actually _in_ the serialized assignment dictionary, in particular, the fields that the frontend will make use of, and the inclusion of a serialized action.

**Review by iloveagent57 — COMMENTED** [review]
One more unit test request, then this is good to go!

**Comment by irfanuddinahmad** [comment]
> One more unit test request, then this is good to go!

@iloveagent57 Could I request a re-review and merge (if all ok)?

**Review by iloveagent57 — APPROVED** [review]

**Pull request merged**